src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -585,4 +616,82 @@ private String mapToType(Reason reason) throws CTPException {
         throw new CTPException(Fault.SYSTEM_ERROR, "Unexpected refusal reason: %s", reason);
     }
   }
+
+  /**
+   * Make Case Service request to return cases by UPRN
+   *
+   * @param uprn of requested cases
+   * @param listCaseEvents boolean of whether require case events
+   * @return List of cases for UPRN
+   * @throws CTPException
+   */
+  private List<CaseDTO> callCaseSvcByUPRN(Long uprn, Boolean listCaseEvents) throws CTPException {
+
+    List<CaseContainerDTO> rmCases = new ArrayList<>();
+    try {
+      rmCases = caseServiceClient.getCaseByUprn(uprn, listCaseEvents);
+    } catch (ResponseStatusException ex) {
+      if (ex.getStatus() == HttpStatus.NOT_FOUND) {
+        log.with(uprn).info("Case by UPRN Not Found calling Case Service");
```

Comment:
The HttpStatus.NOT_FOUND currently occurs when the getCaseByUprn endpoint is called with an invalid uprn (e.g. http://localhost:8171/cases/uprn/1111111111) and there is an existing cuc test for this, which tests for the 404 error to occur. Since it now fails with a 500 error instead the cuc test now fails (I've found this locally and in dev). So I think this need to handle invalid uprns with a 404 again if possible.

Reply:
The functionality has changed Ella. It does handle HTTPStatus.NOT_FOUND in the code. It is not considered an invalid UPRN now if we simply cannot get any cases to return from the Case service. It looks in the Firestore repository for a new skeleton case, then if none is found it requests the address from from AIMs. If the UPRN returns an address from AIMs it saves this to the Firestore repository and returns the skeleton case details. If it is not in AIMS it returns an Exception as an invalid address. I haven't configured Dev correctly as said to you earlier and don't believe it is managing to query AIMs. It works locally for me and will configure Dev when through this marathon of a review.

Reply:
Martin, the cuc test that I'm referring to is https://collaborate2.ons.gov.uk/jira/secure/Tests.jspa#/testCase/CR-T137
It already exists in the test environment. When I run it locally against your branch then I find that it fails.

Reply:
Ella, as said the functionality has changed and so presumably has what is meant by invalid UPRN. If a UPRN is not linked to a case we will create a new skeleton case and publish a NewAddressReported event to RM, store the case in Firestore. Changing the CUC test is presumably what you are working on to the reflect the new functionality? Dev isn't configured correctly yet so it will fail until configured properly. Still working on reviews. So far I have a team record of 4 reviewers, things have often been delivered without review before. This works in my environment but now queries AIMs and uses Firestore,  so config needs to be got rightful these new parts.

Reply:
Martin, if what is meant by an invalid UPRN has now changed then the existing Cucumber test, which tests that an invalid UPRN (now presumably a valid UPRN) will throw a 404 error, will find that that is no longer the case. Therefore the test will fail until it has been amended to reflect this.